pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/participant/ParticipantWrapper.java

```diff
@@ -248,9 +248,18 @@ private void addWrapperToList(ElasticSearchParticipantDto elasticSearchParticipa
 
             List<MedicalRecord> medicalRecord = esDsm.getMedicalRecord();
             List<OncHistoryDetail> oncHistoryDetails = esDsm.getOncHistoryDetail();
+            if (oncHistoryDetails != null) {
+                oncHistoryDetails.removeIf(oncHistoryDetail -> oncHistoryDetail.isDeleted() == true);
+            }
             List<SomaticResultUpload> somaticResultUpload = esDsm.getSomaticResultUpload();
             List<Tissue> tissues = esDsm.getTissue();
+            if (tissues != null) {
+                tissues.removeIf(tissue -> tissue.isDeleted() == true);
+            }
             List<SmId> smIds = esDsm.getSmId();
+            if (smIds != null) {
+                smIds.removeIf(smId -> smId.isDeleted() == true);
+            }
```

Comment:
@zyme @denniscunningham thank you for all of your comments and review! While I'm working on them, do you have any concerns regarding these changes? I added them based on my conversation with Zim, where we talked about having a guard in the backend not sending deleted data to the frontend (instead of frontend having all the control of what to show and what to hide from user). 

Based on the java docs, the `removeIf` is O(n) and at one time DSM will only have one participant's data to process in the `elasticSearchParticipantDto`. However I would love your feedback about if this is a good change in prod based on the current environment settings and the usage.

Alternatives to this are : 
1. not adding deleted data to the ES from the beginning ( changing the upsert and the export code and rebuilding the indices) 
2. changing all of the filtering and query classes to add the condition check to all of the queries we send to ES. This is very scary and would be a big change and bigger to test.

I appreciate your feedback on this

Reply:
@pegahtah Are you talking about a specific commit, or just the code in ParticipantWrapper?

Reply:
@denniscunningham specifically this code in the `ParticipantWrapper` class. streaming the array of tissues and onc histories and removing them on the spot if they are deleted

Reply:
Makes sense to me, and I agree that it is better to do this here since we have no cases where the frontend wants to deal with deleted ptp data.
I'm not worried about the performance. The lists should be small, and I'd bet any perf issues we have would swamp whatever this adds.